lib/ret_web/channels/hub_channel.ex

```diff
@@ -455,6 +456,84 @@ defmodule RetWeb.HubChannel do
     {:noreply, socket}
   end
 
+  def handle_in("list_entity_states", _, socket) do
+    hub = socket |> hub_for_socket
+    entity_states = EntityState.list_entity_states(hub.hub_id)
+    {:reply, {:ok, EntityStateView.render("index.json", %{entity_states: entity_states})}, socket}
+  end
+
+  def handle_in(
+        "create_entity_state",
+        %{"nid" => nid, "create_message" => create_message, "updates" => updates},
+        socket
+      ) do
+    with {:ok, hub} <- authorize(socket, :write_entity_state),
+         {:ok, entity_state} <-
+           EntityState.create_entity_state!(hub, %{
+             nid: nid,
+             create_message: create_message,
+             updates: updates
+           }) do
+      broadcast!(
+        socket,
+        "entity_state_created",
+        EntityStateView.render("show.json", %{entity_state: entity_state})
+      )
+
+      {:reply, :ok, socket}
+    else
```

Comment:
I would benefit from seeing a demonstration so I can understand how to avoid nested `case` or `if` statements. (`authorize/2` seems like a good candidate). I suspect I'm missing some basic elixir knowledge

Reply:
I tried to find some information online to understand why `with-else` is an anti-pattern. From my understanding, it's not `with-else` that is the anti-pattern, but having a complex `else` (e.g. by pattern matching on error tuples). The reason this is a code smell is because it's impossible to tell which clause of the `with` returned the error tuple.

José Valim gives an example and explanation here: https://github.com/lucasvegi/Elixir-Code-Smells#complex-else-clauses-in-with

The suggested remedy is "instead of concentrating all error handlings within a single complex else block, it is better to normalize the return types in specific private functions". I'll do that here.

Reply:
Ok, now each special function returns normalized return types (`{:error, reason}`). I still have an `else` because I want to send the error to the client in a reply, but there's no pattern matching in the `else`, so I think the code smell has been mitigated.